File: packages/dd-trace/test/llmobs/parity/scenarios/openai/responses.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('openai')
  const OpenAI = loadVersionedModule('openai')
  const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY, baseURL: `${process.env.PROVIDER_BASE_URL}/v1` })
  await client.responses.create({
    model: 'gpt-4o-mini',
    input: 'Say parity',
    instructions: 'Be brief',
    max_output_tokens: 16,
    temperature: 0,
  })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/openai/responses.py
import os

from openai import OpenAI

from common import finish, start


start("openai")
client = OpenAI(api_key=os.environ["OPENAI_API_KEY"], base_url=f'{os.environ["PROVIDER_BASE_URL"]}/v1')
client.responses.create(
    model="gpt-4o-mini",
    input="Say parity",
    instructions="Be brief",
    max_output_tokens=16,
    temperature=0,
)
finish()
